### PR TITLE
chore(core): Expose the editableAsText attribute, even if the metadata file doesn't have it yet #27393

### LINKED ISSUE #27393: Edit Content: Expose the `editableAsText` even if the metadata file doesn't have it yet
### Parent Issue

#26910

### Problem Statement

The new `editableAsText` property is available to the REST endpoints as long as the File Asset in dotCMS has been either re-saved or re-indexed before accessing it. This may represent a performance issue in customers with hundreds of thousands of files, and we need the property to be available right away.

### Steps to Reproduce

1. Start up dotCMS version 23.10 or lower. Just make sure that it was created before December 18th, 2023.
2. Use the Full Starter to have files you can test with. Let dotCMS start up and make sure it looks ok.
3. Shut it down, and re-start it using the latest master image.
5. Open up the `/application/vtl/activities/activities-detail.vtl` file.
6. You'll see that the File Asset will NOT display the code editor for you to edit its contents directly, which is wrong. **IMPORTANT: DO NOT click `Save` or `Publish` as this will make the issue non-reproducible.**

### Acceptance Criteria

dotCMS must be able to figure out whether the selected file can be editable directly in the back-end or not whitout relying exclusively on recently indexed metadata. If it can, then its contents must be displayed in a Code Editor field.

### dotCMS Version

Latest master.

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.slack.com/archives/CQQMX585N/p1705592078210759

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
### Proposed Changes
* Including Code Review feedback from Will.
* Reporting via both the UI and the `dotcms.log` file when a File Asset is missing its binary file from the dotCMS Assets folder.
* Fixing a problem in the `compareTo()` method in the `Role` class that is only reproducible when starting up using the generated Starter from our [authoring](https://authoring.dotcms.com/c) instance.
* Adding missing Javadoc and more status/error logging.

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

**Review by dsolistorres — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/business/Role.java [hunk 1]
```diff
@@ -163,34 +171,33 @@ public void setLocked(boolean locked) {
...
+	 * This is because Roles are hierarchical, and the parents need to be imported before the
+	 * children can be imported.
+	 *
+	 * @param role The {@link Role} object to be compared.
+	 *
+	 * @return If this object goes first, returns -1. If the Role object goes first, returns 1. If
+	 * they are equal, returns 0.
+	 */
+	@Override
+	public int compareTo(final Role role) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/business/Role.java:185]: Let's include a test that goes through the `compareTo`

**Review by dsilvam — APPROVED** [review]

**Comment by dsilvam** [comment]
Great job @jcastro-dotcms !

File: dotCMS/src/main/java/com/dotcms/storage/FileStorageAPIImpl.java [hunk 2]
```diff
@@ -418,10 +419,14 @@ public Map<String, Serializable> retrieveMetaData(final FetchMetadataParams requ
      * @param metadataMap The File's metadata Map.
      */
     private void checkEditableAsText(final Map<String, Serializable> metadataMap) {
-        if (!metadataMap.containsKey(BasicMetadataFields.EDITABLE_AS_TEXT.key())) {
-            final String mimeType =
-                    Try.of(() -> metadataMap.get(CONTENT_TYPE_META_KEY.key()).toString()).getOrElse(StringPool.BLANK);
-            metadataMap.put(EDITABLE_AS_TEXT.key(), FileUtil.isFileEditableAsText(mimeType));
+        if (UtilMethods.isSet(metadataMap)) {
```
[the remark this answers is not in the document]
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/storage/FileStorageAPIImpl.java:422]: Done!

File: dotCMS/src/main/java/com/dotmarketing/business/Role.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/business/Role.java:185]: Done!

**Pull request merged**